packages/mobile/src/components/inbox-unavailable-drawer/InboxUnavailableDrawer.tsx

```diff
@@ -0,0 +1,256 @@
+    color: palette.neutralLight2,
+    textTransform: 'uppercase',
+    lineHeight: typography.fontSize.xl * 1.25
+  },
+  infoContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    gap: spacing(4.5),
+    paddingVertical: spacing(2),
+    paddingHorizontal: spacing(4),
+    backgroundColor: palette.neutralLight9,
+    borderWidth: 1,
+    borderColor: palette.neutralLight7,
+    borderRadius: spacing(2)
+  },
+  callToActionText: {
+    color: palette.neutral,
+    fontSize: typography.fontSize.large,
+    fontFamily: typography.fontByWeight.medium,
+    lineHeight: typography.fontSize.large * 1.3,
+    textAlign: 'center'
+  },
+  button: {
+    padding: spacing(4),
+    height: spacing(12)
+  },
+  buttonText: {
+    color: palette.neutral,
+    fontSize: typography.fontSize.large,
+    fontFamily: typography.fontByWeight.bold
+  },
+  buttonTextWhite: {
+    color: palette.white
+  },
+  border: {
+    borderBottomWidth: 1,
+    borderBottomColor: palette.neutralLight8
+  }
+}))
+
+const mapActionToContent = (
+  callToAction: ChatPermissionAction,
+  user: User,
+  styles: ReturnType<typeof useStyles>
+) => {
+  switch (callToAction) {
+    case ChatPermissionAction.NONE:
+      return (
+        <>
+          {messages.noAction}
+          <UserBadges
+            user={user}
+            nameStyle={styles.callToActionText}
+            as={Text}
+          />
+        </>
+      )
+    case ChatPermissionAction.TIP:
+      return (
+        <>
+          {messages.tipGated(
+            <UserBadges
+              user={user}
+              nameStyle={styles.callToActionText}
+              as={Text}
+            />
+          )}
+        </>
+      )
+    case ChatPermissionAction.UNBLOCK:
+      return <>{messages.blockee}</>
+    default:
+      return null
+  }
+}
+
+export const InboxUnavailableDrawer = () => {
+  const dispatch = useDispatch()
+  const navigation = useNavigation()
+  const styles = useStyles()
+  const neutralLight2 = useColor('neutralLight2')
+
+  const { userId } = useSelector((state) => getData<'InboxUnavailable'>(state))
+  const user = useSelector((state) => getUser(state, { id: userId }))
+  const { callToAction } = useSelector((state) => getCanChat(state, userId))
+
+  const closeDrawer = useCallback(() => {
+    dispatch(
+      setVisibility({
+        drawer: 'InboxUnavailable',
+        visible: false
+      })
+    )
+  }, [dispatch])
+
+  const handleUnblockPress = useCallback(() => {
+    dispatch(unblockUser({ userId }))
+    closeDrawer()
+  }, [dispatch, userId, closeDrawer])
+
+  const handleLearnMorePress = useCallback(() => {
+    // TODO: Link to blog
+    closeDrawer()
+  }, [closeDrawer])
+
+  const handleTipPress = useCallback(() => {
+    dispatch(beginTip({ user, source: 'profile' }))
+    navigation.navigate('TipArtist')
+    closeDrawer()
+  }, [closeDrawer, dispatch, navigation, user])
+
+  const actionToButtonsMap = useMemo(() => {
```

Comment:
Thoughts on this vs web impl?

Reply:
Was using web impl but now that we have 2 buttons for unblock case it didn't work as well so changed to this instead. I think you'll have to update web modal for that case too.

Reply:
thanks for the callout, missed that!

In that case, feels like we don't really save much from having these individual props vs just button components, but I'm down for keeping as is too

Reply:
Yeah I forgot I can return a component with `() =>` honestly, would be down for that.

Reply:
Chatted with Marcus on slack, gonna keep it as it is for now. Realized it's weird to pass in the onPress callbacks to `mapActionToContent`, would have to change it to a hook.